src/ISA/ISA/ArcTypes/ArcStudy.fs

```diff
@@ -3,29 +3,38 @@
 open Fable.Core
 open ISA.Aux
 
+module ArcStudyAux =
+    module SanityChecks = 
+        let inline validateUniqueAssayIdentifier (assay: ArcAssay) (existingAssays: seq<ArcAssay>) =
+            match Seq.tryFindIndex (fun x -> x.FileName = assay.FileName) existingAssays with
+            | Some i ->
+                failwith $"Cannot create assay with name {assay.FileName}, as assay names must be unique and assay at index {i} has the same name."
+            | None ->
+                ()
+
 [<AttachMembers>]
 type ArcStudy = 
     {
         ID : URI option
-        FileName : string option
+        mutable FileName : string option
         Identifier : string option
-        Title : string option
-        Description : string option
-        SubmissionDate : string option
-        PublicReleaseDate : string option
-        Publications : Publication list option
-        Contacts : Person list option
-        StudyDesignDescriptors : OntologyAnnotation list option
-        Materials : StudyMaterials option
+        mutable Title : string option
+        mutable Description : string option
+        mutable SubmissionDate : string option
+        mutable PublicReleaseDate : string option
+        mutable Publications : Publication list
+        mutable Contacts : Person list
```

Comment:
Check, if `.Copy()` needs to be adjusted with these now mutable fields.

Reply:
Yes! This will be part of #126